WalletCore/Classes/ViewModel/AssetDisclaimerViewModel.swift

```diff
@@ -0,0 +1,126 @@
+            .mapToNextDisclaimer(fromDisclaimers: disclaimers.value)
+        
+        let disclaimerDeclineRequest = decline.asObservable()
+            .flatMapLatest{
+                authManager.assetDisclaimerDecline.request(withParams: $0)
+            }
+            .shareReplay(1)
+        
+        disclaimer = Driver.merge([
+            disclaimers.asObservable().mapToFirst(),
+            nextDisclaimerToAccept.asDriver(onErrorJustReturn: nil).filterNil()
+        ])
+        
+        dismissViewController =
+            Observable.merge(
+                disclaimerDeclineRequest.filterSuccess().map{ _ in () },
+                nextDisclaimerToAccept.filter{ $0 == nil }.map{ _ in () }
+            )
+            .asDriver(onErrorJustReturn: ())
+        
+        loadingViewModel = LoadingViewModel([
+            disclaimerRequest.isLoading(),
+            disclaimerAcceptRequest.isLoading(),
+            disclaimerDeclineRequest.isLoading()
+        ])
+        
+        
+        /// Bindings
+        disclaimerRequest
+            .filterSuccess()
+            .bind(to: disclaimers)
+            .disposed(by: disposeBag)
+    }
+}
+
+fileprivate extension ObservableType where Self.E == LWRxAuthManagerAssetDisclaimersApprove.Result {
+    
+    /// <#Description#>
+    ///
+    /// - Parameter disclaimers: <#disclaimers description#>
+    /// - Returns: <#return value description#>
+    func mapToNextDisclaimer(fromDisclaimers disclaimers: [LWModelAssetDisclaimer]) -> Observable<LWModelAssetDisclaimer?> {
+        return filterSuccess().map{ disclaimer -> LWModelAssetDisclaimer? in
+            guard let indexOfAcceptedDisclaimer = (disclaimers.index{ $0.id == disclaimer }) else {
+                return nil
+            }
+            
+            return disclaimers[opt: indexOfAcceptedDisclaimer + 1 ]
+        }
+    }
+}
+
+fileprivate extension ObservableType where Self.E == AssetDisclaimerId {
+    
+    /// <#Description#>
+    ///
+    /// - Parameters:
+    ///   - acceptEnabled: <#acceptEnabled description#>
+    ///   - authManager: <#authManager description#>
+    /// - Returns: <#return value description#>
+    func accept(isAcceptEnabled acceptEnabled: Driver<Bool>, authManager: LWRxAuthManager) -> Observable<LWRxAuthManagerAssetDisclaimersApprove.Result>  {
+        return
+            withLatestFrom(acceptEnabled) { (disclaimer: $0, enabled: $1) }
+            .filter{ $0.enabled }
+            .map{ $0.disclaimer }
+            .flatMapLatest{ disclaimerId in
+                authManager.assetDisclaimerAccept.request(withParams: disclaimerId)
+            }
+            .shareReplay(1)
+    }
+}
+
+fileprivate extension ObservableType where Self.E == [LWModelAssetDisclaimer] {
+    
+    /// <#Description#>
+    ///
+    /// - Returns: <#return value description#>
+    func mapToFirst() -> Driver<LWModelAssetDisclaimer> {
+        return map{ $0.first }
+            .asDriver(onErrorJustReturn: nil)
```

Comment:
Instead of ```.asDriver(onErrorJustReturn: nil).filterNil()``` you can call ```.asDriver(onErrorJustReturn: .empty())``` to complete the sequence, and skip the optional filtering.